textalyze.js

```diff
@@ -74,6 +78,45 @@ function getPrintStatistics(map) {
   return output;
 }
 
+/**
+ * Given an input Map, gets the histogram text to print it to the user.
+ * @param {Map} map - The map to be printed to the user.
+ */
+function getHistogramPrintStatistics(map) {
+  if (typeof map !== 'object' || !(map instanceof Map)) {
+    throw new TypeError('The item frequencies must be a valid map.');
+  }
+
+  /**
+   * Given a percent size, creates a bar for displayal in the histogram.
+   * @param {Number} lengthPercent - The size of the bar as a percent value.
+   * @returns {String} bar - The bar string.
+   */
+  function makeHistogramBar(lengthPercent) {
```

Comment:
Sometimes nesting functions like this makes sense, but I don't think it does here.  It uses nothing from the containing scope.